nomos-core/src/da/mod.rs

```diff
@@ -5,20 +5,23 @@ use crate::da::attestation::Attestation;
 use crate::da::blob::Blob;
 use crate::da::certificate::Certificate;
 
+use self::blob::BlobAuth;
+
 pub mod attestation;
 pub mod blob;
 pub mod certificate;
 
 pub trait DaProtocol {
     type Blob: Blob;
+    type Auth: BlobAuth;
     type Attestation: Attestation;
     type Certificate: Certificate;
     type Settings: Clone;
 
     // Construct a new instance
     fn new(settings: Self::Settings) -> Self;
     /// Encode bytes into blobs
-    fn encode<T: AsRef<[u8]>>(&self, data: T) -> Vec<Self::Blob>;
+    fn encode<T: AsRef<[u8]>>(&self, auth: Self::Auth, data: T) -> Vec<Self::Blob>;
```

Comment:
I think this is part of the message itself not the blob per se, unless I am missing something.